internal/uploader/extract_test.go

```diff
@@ -0,0 +1,176 @@
+					events, err := ExtractEvents(buffer)
+
+					Expect(err).NotTo(HaveOccurred())
+					Expect(len(events)).To(Equal(1))
+					Expect(events[0]).To(Equal(generateExpectedEvents(clusterID.String(), config)))
+				})
+			})
+
+			When("the config is complete", func() {
+				BeforeEach(func() {
+					config = Config{
+						DeploymentType:         "depType",
+						DeploymentVersion:      "depVersion",
+						AssistedServiceVersion: "abcdef",
+						Versions: versions.Versions{
+							SelfVersion:     "1.2.3",
+							AgentDockerImg:  "4.5.6",
+							InstallerImage:  "7.8.9",
+							ControllerImage: "a.b.c",
+						},
+					}
+
+					metadataFile(tw, &clusterID, config)
+
+					Expect(tw.Close()).To(Succeed())
+					Expect(gz.Close()).To(Succeed())
+				})
+
+				It("should find this single metadata information for this cluster", func() {
+					events, err := ExtractEvents(buffer)
+
+					Expect(err).NotTo(HaveOccurred())
+					Expect(len(events)).To(Equal(1))
+					Expect(events[0]).To(Equal(generateExpectedEvents(clusterID.String(), config)))
+				})
+			})
+		})
+
+		Context("of 2 different clusters", func() {
+			clusterID1 := strfmt.UUID("cluster1")
+			clusterID2 := strfmt.UUID("cluster2")
+
+			var config1, config2 Config
+
+			When("the configs are complete", func() {
+				BeforeEach(func() {
+					config1 = Config{
+						DeploymentType:         "SaaS",
+						DeploymentVersion:      "1.1.1",
+						AssistedServiceVersion: "latest",
+						Versions: versions.Versions{
+							SelfVersion:     "quay.io/app-sre/assisted-service:latest",
+							AgentDockerImg:  "quay.io/edge-infrastructure/assisted-installer-agent:latest",
+							InstallerImage:  "quay.io/edge-infrastructure/assisted-installer:latest",
+							ControllerImage: "quay.io/edge-infrastructure/assisted-installer-controller:latest",
+						},
+					}
+
+					config2 = Config{
+						DeploymentType:         "SaaS",
+						DeploymentVersion:      "Unknown",
+						AssistedServiceVersion: "2455b9bb394f6fa896131e2a97b34b8b53a23653",
+						Versions: versions.Versions{
+							SelfVersion:     "quay.io/app-sre/assisted-service:2455b9bb394f6fa896131e2a97b34b8b53a23653",
+							AgentDockerImg:  "registry.redhat.io/rhai-tech-preview/assisted-installer-agent-rhel8:v1.0.0-315",
+							InstallerImage:  "registry.redhat.io/rhai-tech-preview/assisted-installer-rhel8:v1.0.0-340",
+							ControllerImage: "registry.redhat.io/rhai-tech-preview/assisted-installer-reporter-rhel8:v1.0.0-418",
+						},
+					}
+
+					metadataFile(tw, &clusterID1, config1)
+					metadataFile(tw, &clusterID2, config2)
+
+					Expect(tw.Close()).To(Succeed())
+					Expect(gz.Close()).To(Succeed())
+				})
+
+				It("should find both metadata information for each cluster", func() {
+					events, err := ExtractEvents(buffer)
+
+					Expect(err).NotTo(HaveOccurred())
+					Expect(len(events)).To(Equal(2))
+					Expect(events).To(Equal([]eventModels.Events{
```

Comment:
Using slice equality checks order and I think this just caused a test flake locally for me:

```
• Failure [0.001 seconds]
Trying to extract events
/home/ncarboni/Source/assisted-service/internal/uploader/extract_test.go:15
  when the reader contains valid data
  /home/ncarboni/Source/assisted-service/internal/uploader/extract_test.go:56
    of 2 different clusters
    /home/ncarboni/Source/assisted-service/internal/uploader/extract_test.go:123
      when the configs are complete
      /home/ncarboni/Source/assisted-service/internal/uploader/extract_test.go:129
        should find both metadata information for each cluster [It]
        /home/ncarboni/Source/assisted-service/internal/uploader/extract_test.go:162

        Expected
            <[]models.Events | len:2, cap:2>: [
                {
                    ClusterID: "cluster2",
                    Metadata: {
                        AssistedInstallerServiceVersion: "quay.io/app-sre/assisted-service:2455b9bb394f6fa896131e2a97b34b8b53a23653",
                        DiscoveryAgentVersion: "registry.redhat.io/rhai-tech-preview/assisted-installer-agent-rhel8:v1.0.0-315",
                        AssistedInstallerVersion: "registry.redhat.io/rhai-tech-preview/assisted-installer-rhel8:v1.0.0-340",
                        AssistedInstallerControllerVersion: "registry.redhat.io/rhai-tech-preview/assisted-installer-reporter-rhel8:v1.0.0-418",
                        DeploymentType: "SaaS",
                        DeploymentVersion: "Unknown",
                        GitRef: "2455b9bb394f6fa896131e2a97b34b8b53a23653",
                    },
                },
                {
                    ClusterID: "cluster1",
                    Metadata: {
                        AssistedInstallerServiceVersion: "quay.io/app-sre/assisted-service:latest",
                        DiscoveryAgentVersion: "quay.io/edge-infrastructure/assisted-installer-agent:latest",
                        AssistedInstallerVersion: "quay.io/edge-infrastructure/assisted-installer:latest",
                        AssistedInstallerControllerVersion: "quay.io/edge-infrastructure/assisted-installer-controller:latest",
                        DeploymentType: "SaaS",
                        DeploymentVersion: "1.1.1",
                        GitRef: "latest",
                    },
                },
            ]
        to equal
            <[]models.Events | len:2, cap:2>: [
                {
                    ClusterID: "cluster1",
                    Metadata: {
                        AssistedInstallerServiceVersion: "quay.io/app-sre/assisted-service:latest",
                        DiscoveryAgentVersion: "quay.io/edge-infrastructure/assisted-installer-agent:latest",
                        AssistedInstallerVersion: "quay.io/edge-infrastructure/assisted-installer:latest",
                        AssistedInstallerControllerVersion: "quay.io/edge-infrastructure/assisted-installer-controller:latest",
                        DeploymentType: "SaaS",
                        DeploymentVersion: "1.1.1",
                        GitRef: "latest",
                    },
                },
                {
                    ClusterID: "cluster2",
                    Metadata: {
                        AssistedInstallerServiceVersion: "quay.io/app-sre/assisted-service:2455b9bb394f6fa896131e2a97b34b8b53a23653",
                        DiscoveryAgentVersion: "registry.redhat.io/rhai-tech-preview/assisted-installer-agent-rhel8:v1.0.0-315",
                        AssistedInstallerVersion: "registry.redhat.io/rhai-tech-preview/assisted-installer-rhel8:v1.0.0-340",
                        AssistedInstallerControllerVersion: "registry.redhat.io/rhai-tech-preview/assisted-installer-reporter-rhel8:v1.0.0-418",
                        DeploymentType: "SaaS",
                        DeploymentVersion: "Unknown",
                        GitRef: "2455b9bb394f6fa896131e2a97b34b8b53a23653",
                    },
                },
            ]

        /home/ncarboni/Source/assisted-service/internal/uploader/extract_test.go:167
```

@pastequo can you make a follow up to fix this test?

Reply:
Same test failed this CI run https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_assisted-service/6427/pull-ci-openshift-assisted-service-master-edge-unit-test/1801653518281478144

Reply:
Yes, I saw it too, and fix in here https://github.com/openshift/assisted-service/pull/6435/files#diff-7eb6290c1b1e1c5de47c84bd0946399c43bb374390d36e89483bfb0a5af7e122

Reply:
:+1: thanks